### PR TITLE
[Merged by Bors] - chore: bump std4 dependency

### PR DESCRIPTION
**Pull request by kim-em**
This is quite a substantial bump as `Nondet` / `backtrack` / `solve_by_elim` have all moved to Std.

---

Infelicities:
* ~~`Except.emoji` has become `private` in Std, so for now I've just made several copies of it here, at each point of use. leanprover/std4#506 makes it public again.~~
* ~~`MVarId.note` is now in `Std.Tactic.SolveByElim`, so we need to (otherwise unnecessarily) import that. leanprover/std4#507 fixes this.~~

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: lake-manifest.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
  [{"url": "https://github.com/leanprover/std4",
    "type": "git",
    "subDir": null,
-   "rev": "d8610e1bcb91c013c3d868821c0ef28bf693be07",
```
Comment (eric-wieser) [inline lake-manifest.json:7]: For reference: https://github.com/leanprover/std4/compare/d8610e1bcb91c013c3d868821c0ef28bf693be07...0f6bc5b32bf5b0498902d3b5f0806c75530539d5

File: Mathlib/Tactic/Relation/Symm.lean [hunk 2]
```diff
@@ -8,42 +8,19 @@ import Std.Tactic.Relation.Symm
...
-end Lean.MVarId
-
 namespace Mathlib.Tactic
 
 open Lean.Elab.Tactic Std.Tactic
 
-/-- For every hypothesis `h : a ~ b` where a `@[symm]` lemma is available,
-add a hypothesis `h_symm : b ~ a`. -/
-elab "symm_saturate" : tactic => liftMetaTactic1 fun g => g.symmSaturate
-
```
Comment (eric-wieser) [inline Mathlib/Tactic/Relation/Symm.lean:46]: Has this been moved to Std, or just deleted? The Std commit log doesn't make any such move obvious.

Comment (semorrison) [reply Mathlib/Tactic/Relation/Symm.lean:46]: This is only used by `solve_by_elim`, and moved to Std as part of https://github.com/leanprover/std4/pull/447

Comment (eric-wieser) [reply Mathlib/Tactic/Relation/Symm.lean:46]: Ah sorry, I was foiled by "load diff" banners blocking by ability to Ctrl+F!

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**